tests/pipelines/test_pipelines_common.py

```diff
@@ -861,6 +873,77 @@ def test_cfg(self):
 
         assert out_cfg.shape == out_no_cfg.shape
 
+    def test_callback_inputs(self):
```

Comment:
Hmm I think we have three types of cases to check here. 

1. Does the it work on a subset of `pipe._callback_tensor_inputs` 
2  Does it work when the all tensor inputs are provided 
3. Does it update the tensor 

```suggestion
    def test_callback_inputs(self):
        sig = inspect.signature(self.pipeline_class.__call__)

        if not ("callback_on_step_end_tensor_inputs" in sig.parameters and "callback_on_step_end" in sig.parameters):
            return

        components = self.get_dummy_components()
        pipe = self.pipeline_class(**components)
        pipe = pipe.to(torch_device)
        pipe.set_progress_bar_config(disable=None)

        self.assertTrue(
            hasattr(pipe, "_callback_tensor_inputs"),
            f" {self.pipeline_class} should have `_callback_tensor_inputs` that defines a list of tensor variables its callback function can use as inputs",
        )

        def callback_inputs_success(pipe, i, t, callback_kwargs):
            # interate over callback args
            for tensor_name, tensor_value in callback_kwargs.items():
                # check that we're only passing in allowed tensor inputs
                assert tensor_name in pipe._callback_tensor_inputs
                # check that a tensor exists
                assert tensor_value is not None

        inputs = self.get_dummy_inputs(torch_device)

        # Test passing in a subset
        inputs["callback_on_step_end"] = callback_inputs_success
        inputs["callback_on_step_end_tensor_inputs"] = ["latents"]
        inputs["output_type"] = "latent"

        output = pipe(**inputs)[0]

        # Test passing in a everything
        inputs["callback_on_step_end"] = callback_inputs_success
        inputs["callback_on_step_end_tensor_inputs"] = pipe._callback_tensor_inputs
        inputs["output_type"] = "latent"

        output = pipe(**inputs)[0]

        def callback_inputs_change_tensor(pipe, i, t, callback_kwargs):
            last_t = callback_kwargs["timesteps"][-1]
            if t == last_t:
                callback_kwargs["latents"] = torch.zeros_like(callback_kwargs["latents"])

            return callback_kwargs

        inputs["callback_on_step_end"] = callback_inputs_change_tensor
        inputs["callback_on_step_end_tensor_inputs"] = incorrect_tensor_inputs
        inputs["output_type"] = "latent"

        output = pipe(**inputs)[0]
        assert output.abs().sum() == 0
```

Reply:
Very nice idea to test with the latents being 0! That's a nice aggressive test. We should think about other aggressive tests we can do here